apollo-adapters/src/jvmTest/kotlin/JavaTimeAdaptersTest.kt

```diff
@@ -0,0 +1,99 @@
+import com.apollographql.apollo3.adapter.DateAdapter
+import com.apollographql.apollo3.adapter.JavaInstantAdapter
+import com.apollographql.apollo3.adapter.JavaLocalDateAdapter
+import com.apollographql.apollo3.adapter.JavaLocalDateTimeAdapter
+import com.apollographql.apollo3.adapter.KotlinxInstantAdapter
+import com.apollographql.apollo3.adapter.KotlinxLocalDateAdapter
+import com.apollographql.apollo3.adapter.KotlinxLocalDateTimeAdapter
+import com.apollographql.apollo3.api.Adapter
+import com.apollographql.apollo3.api.CustomScalarAdapters
+import com.apollographql.apollo3.api.json.BufferedSourceJsonReader
+import com.apollographql.apollo3.api.json.buildJsonString
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.atTime
+import kotlinx.datetime.toInstant
+import okio.Buffer
+import org.junit.Test
+import java.time.LocalTime
+import java.time.ZoneOffset
+import kotlin.test.assertEquals
+
+class JavaTimeAdaptersTest {
+  private fun String.jsonReader() = BufferedSourceJsonReader(Buffer().writeUtf8("\"${this}\""))
+
+  private fun <T> Adapter<T>.fromJson(string: String): T {
+    return fromJson(string.jsonReader(), CustomScalarAdapters.Empty)
+  }
+
+  private fun <T> Adapter<T>.toJson(value: T): String {
+    return buildJsonString {
+      toJson(this, CustomScalarAdapters.Empty, value)
+    }.removePrefix("\"")
+        .removeSuffix("\"")
+  }
+
+  @Test
+  fun instant() {
+    var instant = JavaInstantAdapter.fromJson("2010-06-01T22:19:44.475Z")
+    assertEquals(1275430784475, instant.toEpochMilli())
+    assertEquals("2010-06-01T22:19:44.475Z", JavaInstantAdapter.toJson(instant))
+
+    instant = JavaInstantAdapter.fromJson("2010-06-01T23:19:44.475+01:00")
+    assertEquals(1275430784475, instant.toEpochMilli())
```

Comment:
Is it really true that `2010-06-01T22:19:44.475Z` and `2010-06-01T23:19:44.475+01:00` are the same instant `1275430784475`? Or does that mean that the offset is also ignored when parsing?

Reply:
I think it's true? Second one is 22:19 +1? So both are the same milliseconds offset from epoch (dates are hard...)

Reply:
So sorry! Didn't see the `22` vs `23`. Yes it makes sense 👍 .

Too bad about the lost timezone on the writing side, but I don't think it should cause too much trouble.

Reply:
I think it is fine. This is all used to convey "instant" so shouldn't be used to carry a future local date (I have no idea how one would do that)